submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/nfs/NfsFilesService.kt

```diff
@@ -10,30 +13,45 @@ import ebi.ac.uk.io.FileUtils.copyOrReplaceFile
 import ebi.ac.uk.io.FileUtils.getOrCreateFolder
 import ebi.ac.uk.io.FileUtils.moveFile
 import ebi.ac.uk.io.FileUtils.reCreateFolder
-import ebi.ac.uk.io.Permissions
 import ebi.ac.uk.io.RWXR_XR_X
 import ebi.ac.uk.io.RWX______
 import ebi.ac.uk.io.ext.md5
 import ebi.ac.uk.io.ext.notExist
 import ebi.ac.uk.paths.SubmissionFolderResolver
 import mu.KotlinLogging
-import uk.ac.ebi.extended.serialization.service.FileProcessingService
 import java.io.File
 import java.nio.file.attribute.PosixFilePermission
 
 private val logger = KotlinLogging.logger {}
 
 class NfsFilesService(
     private val folderResolver: SubmissionFolderResolver,
-    private val processingService: FileProcessingService,
 ) : FilesService {
-    override fun persistSubmissionFiles(sub: ExtSubmission): ExtSubmission {
-        val subFolder = getOrCreateSubmissionFolder(sub, sub.permissions().folder)
+    override fun preProcessSubmissionFiles(sub: ExtSubmission): FilePersistenceConfig {
+        val permissions = sub.permissions()
+        val subFolder = getOrCreateSubmissionFolder(sub, permissions.folder)
+        val targetFolder = createTempFolder(subFolder, sub.accNo)
 
-        logger.info { "${sub.accNo} ${sub.owner} Processing files of submission ${sub.accNo} on NFS" }
-        val processed = processAttachedFiles(sub, subFolder, createTempFolder(subFolder, sub.accNo), sub.permissions())
-        logger.info { "${sub.accNo} ${sub.owner} Finished processing files of submission ${sub.accNo} on NFS" }
-        return processed
+        return NfsFilePersistenceConfig(subFolder, targetFolder, permissions)
+    }
+
+    override fun persistSubmissionFile(file: ExtFile, config: FilePersistenceConfig): ExtFile {
+        val (subFolder, targetFolder, permissions) = config as NfsFilePersistenceConfig
+        val extFile = file as NfsFile
```

Comment:
I think we should stop doing this and instead  `require(file is NfsFile) { "File processing handler should only handle Nfs File"}` in this way is super clear method and class expections or preconditions.